server/src/tests/get_proof.rs

```diff
@@ -73,9 +76,14 @@ async fn test_get_proof() -> anyhow::Result<()> {
             }
         ]
     });
-    let response: ForkchoiceUpdatedResponseV1 = handle_request(request, &state_channel).await?;
+    let response: ForkchoiceUpdatedResponseV1 =
+        handle_request(request, &state_channel, &app).await?;
     let payload_id = response.payload_id.unwrap();
 
+    state_channel
+        .reserve_many(state_channel.max_capacity())
+        .await?;
```

Comment:
AFAICT from the docs, this returns a permit iterator that should be called to send data, and as we're reserving a full capacity, all direct calls to the channel from there on should fail?

Reply:
It's probably worth adding that explanation as a comment in the code since it is a sort of counter-intuitive usage of the reserve_many function.

@ilchu future messages to the channel do not fail because the permit iterator is dropped immediately, so all the capacity is released back.

Reply:
I see, it is indeed a bit counter-intuitive.